BlueDB/src/main/java/org/bluedb/api/encryption/EncryptionService.java

```diff
@@ -0,0 +1,31 @@
+package org.bluedb.api.encryption;
+
+import java.util.Optional;
+
+public class EncryptionService {
+
+	protected final Optional<EncryptionConfig> encryptionConfig;
+
+	public EncryptionService(EncryptionConfig encryptionConfig) {
+		if (encryptionConfig != null) {
+			this.encryptionConfig = Optional.of(encryptionConfig);
+		} else {
+			this.encryptionConfig = Optional.empty();
+		}
```

Comment:
You can just say `this.encryptionConfig = Optional.ofNullable(encryptionConfig)` and it'll be set to empty automatically if it is null.

Reply:
Good catch, I was thinking ofNullable meant something else. Just updated that.